### PR TITLE
django_client_class fixture to use custom client class

### PR DESCRIPTION
**Pull request by sliverc**
Rest Framework [0] and other Django extensions use their own custom django client.
Instead of overwriting all client fixtures it is desirable to be able to overwrite client class only.

This is analogue to Django SimpleTestCase [1] where client_class can be overridden as well.

[0] http://www.django-rest-framework.org/api-guide/testing/#apiclient
[1] https://docs.djangoproject.com/en/1.11/_modules/django/test/testcases/#SimpleTestCase

### CODE REVIEW

File: pytest_django/fixtures.py [hunk 1]
```diff
@@ -178,13 +178,19 @@ def transactional_db(request, django_db_setup, django_db_blocker):
 
 
 @pytest.fixture()
-def client():
-    """A Django test client instance."""
+def client_class():
```
Comment (blueyed) [inline pytest_django/fixtures.py]: Can we name it `django_client_class`?
New fixtures should have the `django_` prefix (from/via pytest-django, not necessarily from Django).

Comment (sliverc) [inline pytest_django/fixtures.py]: Renamed it.

**Comment by pelme** [comment]
Why not just override the client fixture in your local project conftest.py?

```py

import pytest
from django.test.client import Client

class CustomClient(Client):
    pass

@pytest.fixture
def client():
    return CustomClient()
```

fixtures are meant to be able to be replaced on a project level. This may not be obvious and could be documented, but I think that approach is cleaner/better. Thoughts? :)

**Comment by sliverc** [comment]
After this change it will still be possible to overwrite `client` fixture if desired. Besides `client` fixture there is also `admin_client`. This means if a user uses both fixtures two fixtures need to be overwritten. And in case of `admin_client` it is not just a one liner.

I could also see more use cases for client fixtures e.g. `auth_client` where a client is authenticated but is not admin. Such a fixture could then be based on `django_client_class` as well.

**Comment by sliverc** [comment]
I guess it is:

```python
@pytest.fixture
def django_client_class():
  return CustomClient
```

vs.

```python
@pytest.fixture
def client():
    return CustomClient()

@pytest.fixture
def admin_client(db, admin_user):
    client = CustomClient()
    client.login(username=admin_user.username, password='password')
    return client
```

It is a small advantage in terms of lines. But if you feel like it is not worth the new fixture feel free to close the PR.

File: pytest_django/fixtures.py [hunk 2]
```diff
@@ -175,13 +175,19 @@ def transactional_db(request, django_db_setup, django_db_blocker):
 
 
 @pytest.fixture()
-def client():
-    """A Django test client instance."""
+def django_client_class():
```
Comment (blueyed) [inline pytest_django/fixtures.py]: Could this be session-scoped then (by default)?

**Comment by sliverc** [comment]
Makes total sense. Set scope to session.

**Comment by sliverc** [comment]
Closing this PR as the gain is minimal and there doesn't seem to be an interest in it.

**Pull request closed without merging**